USFMConverter/Core/Render/RenderDocument.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using USFMConverter.Core.ConstantValue;
+using USFMConverter.Core.Data;
+using USFMToolsSharp;
+using USFMToolsSharp.Models.Markers;
+
+namespace USFMConverter.Core.Render
+{
+    public abstract class RenderDocument
```

Comment:
This class name is confusing.  A class name should be a noun that states what it is.  "RenderDocument" is a verb, not a noun, and makes it unclear what this is.  Instead, this should be named something like "RenderableDocument".